src/stravalib/protocol.py

```diff
@@ -89,6 +102,85 @@ def __init__(
             lambda _request_params, _method: None
         )
 
+    def _token_expired(self) -> bool:
+        """Checks if a token has expired or not. Returns True if it's expired.
+
+        Returns
+        -------
+        bool
+            True if token has expired, otherwise returns false
+        """
+        if self.token_expires:
+            if time.time() > self.token_expires:
+                print("Your token has expired; Refreshing it now.")
+                return True
+            else:
+                return False
+        else:
+            logging.warning(
+                "Please make sure you're set client.token_expires if you",
+                " want automatic token refresh to work",
+            )
+            return False
+
+    def refresh_expired_token(self) -> None:
+        """Checks to see if a token has expired and auto refreshes it
+        if the user has setup their environment with the client
+        secret information.
+
+        Returns
+        -------
+        None
+            If all is setup properly, updates and resets the access_token
+            attribute. Otherwise prints a statement letting the user know that
+            they can set that up if they wish or they need to refresh their
+            token.
+
+        """
+        try:
+            # Load the .env file only if `dotenv` is installed
+            # Mypy doesn't seem to like this import?
+            # Am I missing something?
+            from dotenv import load_dotenv  # type: ignore
+
+            load_dotenv()
+            logging.info("Loaded .env file successfully.")
+        except ImportError:
+            logging.debug(
+                "python-dotenv is not installed. Skipping .env file loading."
+                " If you need .env support, please install `python-dotenv`."
+            )
+        # os.environ.get returns strings by default
+        client_id = os.environ.get("CLIENT_ID")
+        client_secret = os.environ.get("CLIENT_SECRET")
+
+        if client_id and client_secret:
```

Comment:
I'm stuck here on typing and how to ensure client_id is an int. it is possible that a user could write the wrong value to their envt or env file. mypy however is giving me errors .